src/de.rs

```diff
@@ -741,6 +749,9 @@ struct EnumAccess<'de, 'document, 'variant> {
     de: &'variant mut DeserializerFromEvents<'de, 'document>,
     name: Option<&'static str>,
     tag: &'document str,
+    /// a flag to do a hack to run visitor.visit_map() instead of visitor.visit_enum() when `is_serde_content_newtype` is true.
+    /// That is required because `visit_enum` is not available in serde internal newtype visitor.
+    has_visited: bool,
```

Comment:
By itself, representing an enumeration as a one-entry map is not a hacker solution, so there is no need to mention it:
```suggestion
    /// When this struct is used as [`MapAccess`], tracks whether the key was already
    /// requested or not. We want to represent enum value as a one-entry map.
    /// When key was requested, this field is set to `true`
    is_key_deserialized: bool,
```